test-runner/test_runner.py

```diff
@@ -190,13 +192,27 @@ def get_test_list(args: dict, tests_yaml: List[dict]):
         except (DockerException, PerfException, YAMLError) as err:
             logging.error(err)
             summary.append([idx + 1, test.name, "FAIL"])
+            json_summary.append(
+                {"Group": unique_identifier, "Test": test.name, "Status": "FAIL"}
+            )
             ERROR = True
             continue
         except KeyboardInterrupt:
             summary.append([idx + 1, test.name, "FAIL"])
+            json_summary.append(
+                {"Group": unique_identifier, "Test": test.name, "Status": "FAIL"}
+            )
             ERROR = True
             break
         summary.append([idx + 1, test.name, "PASS"])
+        json_summary.append(
+            {"Group": unique_identifier, "Test": test.name, "Status": "PASS"}
+        )
+    json_summary_path = f"test-runner-summary-{unique_identifier}.json"
+
+    with open(json_summary_path, "w") as file:
```

Comment:
```suggestion
    with open(json_summary_path, "w", encoding="utf-8") as file:
```
This will fix your lint error.